test_scripts/collect_data.py

```diff
@@ -0,0 +1,52 @@
+import os
+from time import sleep
+import subprocess
+import re
+
+run_time = 10
+reset_time = 5
+pps_sample = [10000,50000,100000]
+pcycle_sample = [20,200,2000]
+num_runs = 3
+
+
+def get_hw_latency(text):
+    pat = re.findall("P99 latency:.*", text)[0]
+    pat = pat.split(':')[-1]
+    return float(pat)
+
+def get_pipline_latency(text):
+    pat = re.findall("percentile_values_ns:.*", text)[0]
+    pat = pat.split(':')[-1]
+    return float(pat)
+
+
+def do_test(i,pps, pcycle):
+    with open("data", 'a') as f:
+        f.write("Run %d --- params: pps=%d pcycle=%d\n"%(i, pps,pcycle))
+    os.system("../bin/bessctl 'run samples/hw_ts_test BESS_PKT_SIZE=200,BESS_PKT_CYCLE="+str(pcycle)+",BESS_PKT_RATE="+str(pps)+"'")
```

Comment:
I'd recomment restart BESS daemon for each run. 'daemon start' and then 'yes' when it asks whether to kill the current daemon.